src/observableApiConfig.ts

```diff
@@ -91,14 +93,15 @@ export async function getDeployConfig(
 
 export async function setDeployConfig(
   sourceRoot: string,
+  requestedDeployConfigPath: string | null = null,
   newConfig: DeployConfig,
   effects: ConfigEffects = defaultEffects
 ): Promise<void> {
-  const dir = op.join(effects.cwd(), sourceRoot, ".observablehq");
-  const deployConfigPath = op.join(dir, "deploy.json");
+  const deployConfigPath =
+    requestedDeployConfigPath ?? op.join(effects.cwd(), sourceRoot, ".observablehq", "deploy.json");
   const oldConfig = (await getDeployConfig(sourceRoot)) || {};
```

Comment:
Given that we already materialized the default `deployConfigPath` at this point, it’s a bit dangerous to not pass it through to `getDeployConfig` — we should pass it in to avoid `getDeployConfig` again computing the default. (A bit paranoid but a good practice.) I’ll make this change.